web/admin/src/App.js

```diff
@@ -21,23 +21,25 @@ export default class App extends Component {
   componentDidMount() {
     fbc.signinAdmin()
     .then(() => {
-      client.getUsers().then(attendees => this.setState({attendees: attendees.sort(sortPlayers)}))
+      client.getUsers().then(attendees => {this.setState({attendees: attendees.sort(sortPlayers)}) 
 
-      const usersRef = fbc.database.public.usersRef()
-      usersRef.on('child_added', data => {
-        this.setState(state => ({players: [...state.players, {...data.val(), id: data.key}].sort(sortPlayers)}))
-      })
-      usersRef.on('child_removed', data => {
-        this.setState(state => ({players: state.players.filter(p => p.id !== data.key)}))
-      })
-
-      fbc.database.public.adminRef('targets').on('value', data => {
-        this.setState({isGameInProgress: !!data.val()})
-      })
-
-      fbc.database.private.adminableUsersRef().on('value', data => {
-        const users = data.val() || {}
-        this.setState({admins: Object.keys(users).filter(id => users[id].adminToken)})
+        const usersRef = fbc.database.public.usersRef()
+        usersRef.on('child_added', data => {
+          var player = attendees.find(newUser => newUser.id === data.val().id)
```

Comment:
Also, `data.val()` will be called many times as it tries to `find()` the player.  You should be able to  substitute `data.val().id` for `data.key` since the key is what is used for the id.

So `var player = attendees.find(a => a.id === data.key)` should be equivalent